Makefile

```
@@ -187,7 +187,7 @@ wheel:
 	for file in gradient quantum _ods_common catalyst mitigation _transform; do \
 		cp $(COPY_FLAGS) $(DIALECTS_BUILD_DIR)/python_packages/quantum/mlir_quantum/dialects/*$${file}* $(MK_DIR)/frontend/mlir_quantum/dialects ; \
 	done
-	cp $(COPY_FLAGS) $(DIALECTS_BUILD_DIR)/python_packages/quantum/mlir_quantum/compiler_driver.so $(MK_DIR)/frontend/mlir_quantum/
```

You need to create the directory before copying something there.

```suggestion
	mkdir -p $(MK_DIR)/frontend/catalyst/build/bin
```

But this is not completely, because in the `get_path_lib` function you are saying that `catalyst-cli` will be located in the lib folder once installed:

```python
def get_lib_path(project, env_var):
    """Get the library path."""
    if INSTALLED:
        return os.path.join(package_root, "..", "lib")  # pragma: no cover
    return os.getenv(env_var, DEFAULT_LIB_PATHS.get(project, ""))
```

So you may either keep copying catalyst-cli to bin and add a new function `get_bin_path` or change the location of catalyst-cli to the lib folder. I prefer the first option.